contracts/cw-ics721-bridge/src/helpers.rs

```diff
@@ -0,0 +1,107 @@
+use cosmwasm_std::{from_binary, to_binary, Binary, CosmosMsg, Deps, SubMsg, WasmMsg};
+use ics721::{
+    Ics721CallbackMsg, Ics721Callbacks, Ics721Memo, Ics721ReceiveMsg, Ics721Status,
+    NonFungibleTokenPacketData,
+};
+use serde::Deserialize;
+
+use crate::ibc::ACK_CALLBACK_REPLY_ID;
+
+/// Parse the memo field into the type we want
+/// Ideally it would be `Ics721Memo` type or any type that extends it
+fn parse_memo<T: for<'de> Deserialize<'de>>(memo: Option<String>) -> Option<T> {
+    let binary = Binary::from_base64(memo?.as_str()).ok()?;
+    from_binary::<T>(&binary).ok()
+}
+
+/// Parse callback from the memo field
+fn parse_callback(memo: Option<String>) -> Option<Ics721Callbacks> {
+    parse_memo::<Ics721Memo>(memo)?.callbacks
+}
+
+// Create a subMsg that execute the callback on the sender callback
+// we use a subMsg on error because we don't want to fail the whole tx
+// if the callback fails
+// if we were to fail the whole tx, the NFT would have been minted on
+// the other chain while the NFT on this chain would not have been
+// burned
+pub(crate) fn ack_callback_msg(
+    deps: Deps,
+    status: Ics721Status,
+    packet: NonFungibleTokenPacketData,
+) -> Option<SubMsg> {
+    // Get the callback object
+    let callbacks = parse_callback(packet.memo.clone())?;
+
+    // Validate the address
+    let contract_addr = deps
+        .api
+        .addr_validate(packet.sender.as_str())
+        .ok()?
+        .to_string();
+
+    // Create the message we send to the contract
+    // The status is the status we want to send back to the contract
+    // The msg is the msg we forward from the sender
+    let msg = to_binary(&ics721::ReceiverExecuteMsg::Ics721Callback(
+        Ics721CallbackMsg {
+            status,
+            msg: callbacks.src_callback_msg?,
+            original_packet: packet,
+        },
+    ))
+    .ok()?;
+
+    Some(SubMsg::reply_on_error(
```

Comment:
This is a "response" to the transfer if it went successful or not, we don't want to fail the whole tx in case the callback fails here.
so if the callback fails, we just continue as normal.